File: 03.programmers/2019_kakao_winter_internship/01.C++/01.cpp
#include<iostream>
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
	int answer = 0;
	vector<int> Stack;
	for (auto col : moves) {
		for (int i = 0; i < board.size(); i++) {
			int tmp = board[i][col - 1];
			if (tmp) {
				board[i][col - 1] = 0;
				if (!Stack.empty() && Stack.back()==tmp) {
					Stack.pop_back();
					answer += 2;	
				}
				else Stack.push_back(tmp);
				break;
			}
		}
	}
	return answer;
}
int main() {
	vector<vector<int>> b = {{0, 0, 0, 0, 0}, 
							 {0, 0, 1, 0, 3}, 
							 {0, 2, 5, 0, 1}, 
							 {4, 2, 4, 4, 2}, 
							 {3, 5, 1, 3, 1}};
	cout << solution(b, { 1,5,3,5,1,2,1,4 }) << endl;
	return 0;
}

File: 2023/Programmers/lv01/javascript/바탕화면정리/01.js
function solution(wallpaper) {
  var answer = [];
  // 행 길이
  const n = wallpaper.length;
  // 열 길이
  const m = wallpaper[0].length;

  // 시작점
  let sy = n,
    sx = m;
  // 끝점
  let ey = 0,
    ex = 0;

  // 바탕화면 탐색
  for (let i = 0; i < n; i++) {
    for (let j = 0; j < m; j++) {
      // 시작점을 찾기위한 정방향 탐색
      if (wallpaper[i][j] === "#") {
        if (sy > i) {
          sy = i;
        }
        if (sx > j) {
          sx = j;
        }
      }
      // 끝점을 찾기위한 역방향 탐색
      if (wallpaper[n - i - 1][m - j - 1] === "#") {
        if (ey <= n - i) {
          ey = n - i;
        }
        if (ex <= m - j) {
          ex = m - j;
        }
      }
    }
  }
  answer = [sy, sx, ey, ex];

  return answer;
}
